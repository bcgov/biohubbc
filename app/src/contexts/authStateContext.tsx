import useKeycloakWrapper from 'hooks/useKeycloakWrapper';
import React from 'react';

export interface IAuthState {
  ready?: boolean;
}

export const AuthStateContext = React.createContext<IAuthState>({
  ready: false
});

export const AuthStateContextProvider: React.FC = (props) => {
  const keycloakWrapper = useKeycloakWrapper();

  return (
    <AuthStateContext.Provider value={{ ready: keycloakWrapper.keycloak?.authenticated }}>
      {props.children}
    </AuthStateContext.Provider>
  );
};
