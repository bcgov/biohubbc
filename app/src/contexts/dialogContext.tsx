import IconButton from '@material-ui/core/IconButton';
import Snackbar from '@material-ui/core/Snackbar';
import CloseIcon from '@material-ui/icons/Close';
import { Color } from '@material-ui/lab/Alert';
import { ErrorDialog, IErrorDialogProps } from 'components/dialog/ErrorDialog';
import YesNoDialog, { IYesNoDialogProps } from 'components/dialog/YesNoDialog';
import React, { createContext, ReactNode, useState } from 'react';

export interface IDialogContext {
  /**
   * Set the yes no dialog props.
   *
   * Note: Any props that are not provided, will default to whatever value was previously set (or the default value)
   *
   * @memberof IDialogContext
   */
  setYesNoDialog: (props: Partial<IYesNoDialogProps>) => void;
  /**
   * The current yes no dialog props.
   *
   * @type {IYesNoDialogProps}
   * @memberof IDialogContext
   */
  yesNoDialogProps: IYesNoDialogProps;
  /**
   * Set the error dialog props.
   *
   * Note: Any props that are not provided, will default to whatever value was previously set (or the default value)
   *
   * @memberof IDialogContext
   */
  setErrorDialog: (props: Partial<IErrorDialogProps>) => void;
  /**
   * The current error dialog props.
   *
   * @type {IErrorDialogProps}
   * @memberof IDialogContext
   */
  errorDialogProps: IErrorDialogProps;
  /**
   * Set the snackbar props.
   *
   * Note: Any props that are not provided, will default to whatever value was previously set (or the default value)
   *
   * @memberof IDialogContext
   */
  setSnackbar: (props: Partial<ISnackbarProps>) => void;
  /**
   * The current snackbar props.
   *
   * @type {ISnackbarProps}
   * @memberof IDialogContext
   */
  snackbarProps: ISnackbarProps;
}

export interface ISnackbarProps {
  open: boolean;
  onClose: () => void;
  severity?: Color;
  color?: Color;
  snackbarMessage: ReactNode;
}

export const defaultYesNoDialogProps: IYesNoDialogProps = {
  dialogTitle: '',
  dialogText: '',
  open: false,
  onClose: () => {
    // default do nothing
  },
  onNo: () => {
    // default do nothing
  },
  onYes: () => {
    // default do nothing
  }
};

export const defaultErrorDialogProps: IErrorDialogProps = {
  dialogTitle: '',
  dialogText: '',
  open: false,
  onClose: () => {
    // default do nothing
  },
  onOk: () => {
    // default do nothing
  }
};

export const defaultSnackbarProps: ISnackbarProps = {
  snackbarMessage: '',
  open: false,
  onClose: () => {
    // default do nothing
  }
};

export const DialogContext = createContext<IDialogContext>({
  setYesNoDialog: () => {
    // default do nothing
  },
  yesNoDialogProps: defaultYesNoDialogProps,
  setErrorDialog: () => {
    // default do nothing
  },
  errorDialogProps: defaultErrorDialogProps,
  setSnackbar: () => {
    // default do nothing
  },
  snackbarProps: defaultSnackbarProps
});

/**
 * Wraps the provided children in a context that provides various modal dialogs/popups.
 *
 * @param {*} props
 * @return {*}
 */
export const DialogContextProvider: React.FC = (props) => {
  const [yesNoDialogProps, setYesNoDialogProps] = useState<IYesNoDialogProps>(defaultYesNoDialogProps);

  const [errorDialogProps, setErrorDialogProps] = useState<IErrorDialogProps>(defaultErrorDialogProps);

  const [snackbarProps, setSnackbarProps] = useState<ISnackbarProps>(defaultSnackbarProps);

  const setYesNoDialog = function (partialProps: Partial<IYesNoDialogProps>) {
    setYesNoDialogProps({ ...yesNoDialogProps, ...partialProps });
  };

  const setSnackbar = function (partialProps: Partial<ISnackbarProps>) {
    setSnackbarProps({ ...snackbarProps, ...partialProps });
  };

  const setErrorDialog = function (partialProps: Partial<IErrorDialogProps>) {
    setErrorDialogProps({ ...errorDialogProps, ...partialProps });
  };

  return (
    <DialogContext.Provider
      value={{
        setYesNoDialog,
        yesNoDialogProps,
        setErrorDialog,
        errorDialogProps,
        setSnackbar,
        snackbarProps
      }}>
      {props.children}
      <YesNoDialog {...yesNoDialogProps} />
      <ErrorDialog {...errorDialogProps} />
      <Snackbar
        anchorOrigin={{
          vertical: 'bottom',
          horizontal: 'center'
        }}
        open={snackbarProps.open}
        autoHideDuration={6000}
        onClose={() => setSnackbar({ open: false })}
        message={snackbarProps.snackbarMessage}
        action={
          <React.Fragment>
            <IconButton size="small" aria-label="close" color="inherit" onClick={() => setSnackbar({ open: false })}>
              <CloseIcon fontSize="small" />
            </IconButton>
          </React.Fragment>
        }
      />
    </DialogContext.Provider>
  );
};
