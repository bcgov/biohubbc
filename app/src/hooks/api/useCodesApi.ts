import { AxiosInstance } from 'axios';
import { IGetAllCodeSetsResponse } from 'interfaces/useCodesApi.interface';

/**
 * Returns a set of supported api methods for working with projects.
 *
 * @param {AxiosInstance} axios
 * @return {*} object whose properties are supported api methods.
 */
const useCodesApi = (axios: AxiosInstance) => {
  /**
   * Fetch all code sets.
   *
   * @return {*}  {Promise<IGetAllCodeSetsResponse>}
   */
  const getAllCodeSets = async (): Promise<IGetAllCodeSetsResponse> => {
    const { data } = await axios.get('/api/codes/');

    return data;
  };

  return {
    getAllCodeSets
  };
};

export default useCodesApi;
