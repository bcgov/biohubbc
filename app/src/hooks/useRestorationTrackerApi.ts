import axios from 'axios';
import useAdminApi from './api/useAdminApi';
import useAxios from './api/useAxios';
import useCodesApi from './api/useCodesApi';
import useDraftApi from './api/useDraftApi';
import useExternalApi from './api/useExternalApi';
import useProjectApi, { usePublicProjectApi } from './api/useProjectApi';
import useSearchApi, { usePublicSearchApi } from './api/useSearchApi';
import useUserApi from './api/useUserApi';
import usePermitApi from './api/usePermitApi';
import { useContext } from 'react';
import { ConfigContext } from 'contexts/configContext';

/**
 * Returns a set of supported api methods.
 *
 * @return {*} object whose properties are supported api methods.
 */
export const useRestorationTrackerApi = () => {
  const config = useContext(ConfigContext);
  const apiAxios = useAxios(config?.API_HOST);

  const project = useProjectApi(apiAxios);

  const permit = usePermitApi(apiAxios);

  const search = useSearchApi(apiAxios);

  const codes = useCodesApi(apiAxios);

  const draft = useDraftApi(apiAxios);

  const user = useUserApi(apiAxios);

  const admin = useAdminApi(apiAxios);

  const external = useExternalApi(axios);

  const publicApis = {
    project: usePublicProjectApi(apiAxios),
    search: usePublicSearchApi(apiAxios)
  };

  return {
    project,
    permit,
    search,
    codes,
    draft,
    user,
    admin,
    external,
    public: publicApis
  };
};
