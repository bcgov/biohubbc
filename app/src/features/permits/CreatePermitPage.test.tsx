import { cleanup, fireEvent, render, waitFor } from '@testing-library/react';
import { useRestorationTrackerApi } from 'hooks/useRestorationTrackerApi';
import { IGetAllCodeSetsResponse } from 'interfaces/useCodesApi.interface';
import React from 'react';
import CreatePermitPage from './CreatePermitPage';
import { MemoryRouter, Router } from 'react-router';
import { createMemoryHistory } from 'history';
import { DialogContextProvider } from 'contexts/dialogContext';

const history = createMemoryHistory();

jest.mock('../../hooks/useRestorationTrackerApi');

const mockuseRestorationTrackerApi = {
  codes: {
    getAllCodeSets: jest.fn<Promise<IGetAllCodeSetsResponse>, []>()
  },
  permit: {
    createPermits: jest.fn()
  }
};

const mockRestorationTrackerApi = ((useRestorationTrackerApi as unknown) as jest.Mock<
  typeof mockuseRestorationTrackerApi
>).mockReturnValue(mockuseRestorationTrackerApi);

describe('CreatePermitPage', () => {
  beforeEach(() => {
    // clear mocks before each test
    mockRestorationTrackerApi().permit.createPermits.mockClear();
    mockRestorationTrackerApi().codes.getAllCodeSets.mockClear();
  });

  afterEach(() => {
    cleanup();
  });

  it('shows circular spinner when codes not yet loaded', async () => {
    const { asFragment } = render(
      <MemoryRouter>
        <CreatePermitPage />
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(asFragment()).toMatchSnapshot();
    });
  });

  it('renders correctly when codes are loaded', async () => {
    mockRestorationTrackerApi().codes.getAllCodeSets.mockResolvedValue({
      coordinator_agency: [{ id: 1, name: 'agency 1' }]
    } as any);

    const { asFragment, getAllByText } = render(
      <MemoryRouter>
        <CreatePermitPage />
      </MemoryRouter>
    );

    await waitFor(() => {
      expect(getAllByText('Create Non-Sampling Permits').length).toEqual(1);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('Are you sure? Dialog', () => {
    it('calls history.push() if the user clicks `Yes`', async () => {
      mockRestorationTrackerApi().codes.getAllCodeSets.mockResolvedValue({
        coordinator_agency: [{ id: 1, name: 'agency 1' }]
      } as any);

      history.push('/home');
      history.push('/admin/permits/create');

      const { getByText, getAllByText, getByTestId } = render(
        <DialogContextProvider>
          <Router history={history}>
            <CreatePermitPage />
          </Router>
        </DialogContextProvider>
      );

      await waitFor(() => {
        expect(getAllByText('Create Non-Sampling Permits').length).toEqual(1);
      });

      fireEvent.click(getByText('Cancel'));

      await waitFor(() => {
        expect(getAllByText('Create Non-Sampling Permits').length).toEqual(1);
        expect(getByText('Are you sure you want to cancel?')).toBeInTheDocument();
      });

      fireEvent.click(getByTestId('yes-button'));

      await waitFor(() => {
        expect(history.location.pathname).toEqual('/admin/permits');
      });
    });

    it('does nothing if the user clicks `No` or away from the dialog', async () => {
      mockRestorationTrackerApi().codes.getAllCodeSets.mockResolvedValue({
        coordinator_agency: [{ id: 1, name: 'agency 1' }]
      } as any);

      const { getAllByText, getByText, getAllByRole, getByTestId } = render(
        <DialogContextProvider>
          <Router history={history}>
            <CreatePermitPage />
          </Router>
        </DialogContextProvider>
      );

      await waitFor(() => {
        expect(getAllByText('Create Non-Sampling Permits').length).toEqual(1);
      });

      fireEvent.click(getByText('Cancel'));

      await waitFor(() => {
        expect(getByText('Cancel Create Permits')).toBeInTheDocument();
        expect(getByText('Are you sure you want to cancel?')).toBeInTheDocument();
      });

      fireEvent.click(getByTestId('no-button'));

      await waitFor(() => {
        expect(getAllByText('Create Non-Sampling Permits').length).toEqual(1);
      });

      fireEvent.click(getByText('Cancel'));

      await waitFor(() => {
        expect(getByText('Cancel Create Permits')).toBeInTheDocument();
        expect(getByText('Are you sure you want to cancel?')).toBeInTheDocument();
      });

      // Get the backdrop, then get the firstChild because this is where the event listener is attached
      //@ts-ignore
      fireEvent.click(getAllByRole('presentation')[0].firstChild);

      await waitFor(() => {
        expect(getAllByText('Create Non-Sampling Permits').length).toEqual(1);
      });
    });
  });
});
