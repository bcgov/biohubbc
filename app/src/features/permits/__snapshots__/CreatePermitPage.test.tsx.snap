// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePermitPage renders correctly when codes are loaded 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-13"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthXl"
    >
      <div
        class="MuiBox-root MuiBox-root-14"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1"
        >
          Create Non-Sampling Permits
        </h1>
      </div>
      <div
        class="MuiBox-root MuiBox-root-15"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Please add any non-sampling permits here. You can later associate them to a project at any time.
        </p>
      </div>
      <div
        class="MuiPaper-root MuiBox-root MuiBox-root-16 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiBox-root MuiBox-root-17"
        >
          <div
            class="MuiBox-root MuiBox-root-18"
          >
            <h2
              class="MuiTypography-root MuiTypography-h2"
            >
              Coordinator Information
            </h2>
            <div
              class="MuiBox-root MuiBox-root-19"
            >
              <p
                class="MuiTypography-root MuiTypography-body2"
              >
                Enter the contact information for the person directly responsible for the permit. This information will be used as the primary contact should questions arise about this permit.
              </p>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-20"
          >
            <form>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                      data-shrink="false"
                      for="first_name"
                      id="first_name-label"
                    >
                      First Name
                      <span
                        aria-hidden="true"
                        class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                      >
                         *
                      </span>
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input"
                        id="first_name"
                        name="first_name"
                        required=""
                        type="text"
                        value=""
                      />
                      <fieldset
                        aria-hidden="true"
                        class="PrivateNotchedOutline-root-22 MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="PrivateNotchedOutline-legendLabelled-24"
                        >
                          <span>
                            First Name *
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                      data-shrink="false"
                      for="last_name"
                      id="last_name-label"
                    >
                      Last Name
                      <span
                        aria-hidden="true"
                        class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                      >
                         *
                      </span>
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input"
                        id="last_name"
                        name="last_name"
                        required=""
                        type="text"
                        value=""
                      />
                      <fieldset
                        aria-hidden="true"
                        class="PrivateNotchedOutline-root-22 MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="PrivateNotchedOutline-legendLabelled-24"
                        >
                          <span>
                            Last Name *
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                      data-shrink="false"
                      for="email_address"
                      id="email_address-label"
                    >
                      Business Email Address
                      <span
                        aria-hidden="true"
                        class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                      >
                         *
                      </span>
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input"
                        id="email_address"
                        name="email_address"
                        required=""
                        type="text"
                        value=""
                      />
                      <fieldset
                        aria-hidden="true"
                        class="PrivateNotchedOutline-root-22 MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="PrivateNotchedOutline-legendLabelled-24"
                        >
                          <span>
                            Business Email Address *
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <div
                    aria-expanded="false"
                    class="MuiAutocomplete-root MuiAutocomplete-hasClearIcon"
                    data-testid="coordinator_agency"
                    role="combobox"
                  >
                    <div
                      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                        data-shrink="false"
                        for="coordinator_agency"
                        id="coordinator_agency-label"
                      >
                        Contact Agency
                        <span
                          aria-hidden="true"
                          class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                        >
                           *
                        </span>
                      </label>
                      <div
                        class="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                      >
                        <input
                          aria-autocomplete="list"
                          aria-invalid="false"
                          autocapitalize="none"
                          autocomplete="off"
                          class="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                          id="coordinator_agency"
                          required=""
                          spellcheck="false"
                          type="text"
                          value=""
                        />
                        <div
                          class="MuiAutocomplete-endAdornment"
                        >
                          <button
                            aria-label="Clear"
                            class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator MuiAutocomplete-clearIndicatorDirty"
                            tabindex="-1"
                            title="Clear"
                            type="button"
                          >
                            <span
                              class="MuiIconButton-label"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                                />
                              </svg>
                            </span>
                            <span
                              class="MuiTouchRipple-root"
                            />
                          </button>
                        </div>
                        <fieldset
                          aria-hidden="true"
                          class="PrivateNotchedOutline-root-22 MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            class="PrivateNotchedOutline-legendLabelled-24"
                          >
                            <span>
                              Contact Agency *
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root MuiBox-root-26"
              >
                <fieldset
                  class="MuiFormControl-root"
                >
                  <legend
                    class="MuiBox-root MuiBox-root-27 makeStyles-legend-21"
                  >
                    Share Contact Details
                  </legend>
                  <p
                    class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
                  >
                    Do you want the project contact’s name and email address visible to the public?
                  </p>
                  <div
                    class="MuiBox-root MuiBox-root-28"
                  >
                    <div
                      aria-label="Share Contact Details"
                      class="MuiFormGroup-root"
                      role="radiogroup"
                    >
                      <label
                        class="MuiFormControlLabel-root"
                      >
                        <span
                          aria-disabled="false"
                          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-29 MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-checked-30 Mui-checked MuiIconButton-colorPrimary"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <input
                              checked=""
                              class="PrivateSwitchBase-input-32"
                              name="share_contact_details"
                              required=""
                              type="radio"
                              value="false"
                            />
                            <div
                              class="PrivateRadioButtonIcon-root-33 PrivateRadioButtonIcon-checked-35"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                                />
                              </svg>
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-34 MuiSvgIcon-fontSizeSmall"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                                />
                              </svg>
                            </div>
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </span>
                        <span
                          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                        >
                          No
                        </span>
                      </label>
                      <label
                        class="MuiFormControlLabel-root"
                      >
                        <span
                          aria-disabled="false"
                          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-29 MuiRadio-root MuiRadio-colorPrimary MuiIconButton-colorPrimary"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <input
                              class="PrivateSwitchBase-input-32"
                              name="share_contact_details"
                              required=""
                              type="radio"
                              value="true"
                            />
                            <div
                              class="PrivateRadioButtonIcon-root-33"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                                />
                              </svg>
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-34 MuiSvgIcon-fontSizeSmall"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                                />
                              </svg>
                            </div>
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </span>
                        <span
                          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                        >
                          Yes
                        </span>
                      </label>
                      <p
                        class="MuiFormHelperText-root Mui-required"
                      />
                    </div>
                  </div>
                </fieldset>
              </div>
            </form>
          </div>
        </div>
        <hr
          class="MuiDivider-root makeStyles-sectionDivider-12"
        />
        <div
          class="MuiBox-root MuiBox-root-36"
        >
          <div
            class="MuiBox-root MuiBox-root-37"
          >
            <h2
              class="MuiTypography-root MuiTypography-h2"
            >
              Non-Sampling Permits
            </h2>
            <div
              class="MuiBox-root MuiBox-root-38"
            >
              <p
                class="MuiTypography-root MuiTypography-body2"
              >
                Enter any scientific collection, wildlife act and/or park use permits. Provide the last 6 digits of the permit number. The last 6 digits are those after the hyphen (e.g. for KA12-845782 enter 845782).
              </p>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-39"
          >
            <form>
              <div
                class="MuiBox-root MuiBox-root-40"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                  >
                    <div
                      class="MuiBox-root MuiBox-root-41"
                    >
                      <div
                        class="MuiBox-root MuiBox-root-42"
                      >
                        <div
                          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                        >
                          <label
                            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                            data-shrink="false"
                            for="permits.[0].permit_number"
                            id="permits.[0].permit_number-label"
                          >
                            Permit Number
                            <span
                              aria-hidden="true"
                              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                            >
                               *
                            </span>
                          </label>
                          <div
                            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                          >
                            <input
                              aria-invalid="false"
                              class="MuiInputBase-input MuiOutlinedInput-input"
                              id="permits.[0].permit_number"
                              name="permits.[0].permit_number"
                              required=""
                              type="text"
                              value=""
                            />
                            <fieldset
                              aria-hidden="true"
                              class="PrivateNotchedOutline-root-22 MuiOutlinedInput-notchedOutline"
                            >
                              <legend
                                class="PrivateNotchedOutline-legendLabelled-24"
                              >
                                <span>
                                  Permit Number *
                                </span>
                              </legend>
                            </fieldset>
                          </div>
                        </div>
                      </div>
                      <div
                        class="MuiBox-root MuiBox-root-43"
                      >
                        <div
                          class="MuiFormControl-root"
                          style="width: 100%;"
                        >
                          <label
                            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                            data-shrink="false"
                            id="permit_type"
                          >
                            Permit Type
                            <span
                              aria-hidden="true"
                              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                            >
                               *
                            </span>
                          </label>
                          <div
                            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                          >
                            <div
                              aria-haspopup="listbox"
                              aria-label="Permit Type"
                              aria-labelledby="permit_type permits.[0].permit_type"
                              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                              id="permits.[0].permit_type"
                              role="button"
                              tabindex="0"
                            >
                              <span>
                                ​
                              </span>
                            </div>
                            <input
                              aria-hidden="true"
                              class="MuiSelect-nativeInput"
                              name="permits.[0].permit_type"
                              required=""
                              tabindex="-1"
                              value=""
                            />
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M7 10l5 5 5-5z"
                              />
                            </svg>
                            <fieldset
                              aria-hidden="true"
                              class="PrivateNotchedOutline-root-22 MuiOutlinedInput-notchedOutline"
                            >
                              <legend
                                class="PrivateNotchedOutline-legendLabelled-24"
                              >
                                <span>
                                  Permit Type
                                </span>
                              </legend>
                            </fieldset>
                          </div>
                          <p
                            class="MuiFormHelperText-root MuiFormHelperText-contained Mui-required"
                          />
                        </div>
                      </div>
                      <div
                        class="MuiBox-root MuiBox-root-44"
                      >
                        <button
                          aria-label="remove permit"
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
                          data-testid="delete-icon"
                          tabindex="0"
                          type="button"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <svg
                              role="presentation"
                              style="width: 1.5rem; height: 1.5rem;"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M9,3V4H4V6H5V19A2,2 0 0,0 7,21H17A2,2 0 0,0 19,19V6H20V4H15V3H9M7,6H17V19H7V6M9,8V17H11V8H9M13,8V17H15V8H13Z"
                                style="fill: currentColor;"
                              />
                            </svg>
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiBox-root MuiBox-root-45"
                >
                  <button
                    aria-label="add permit"
                    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Add New Permit
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
        <hr
          class="MuiDivider-root makeStyles-sectionDivider-12"
        />
        <div
          class="MuiBox-root MuiBox-root-46"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-actionButton-7 MuiButton-containedPrimary"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Save and Exit
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-actionButton-7 MuiButton-outlinedPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Cancel
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CreatePermitPage shows circular spinner when codes not yet loaded 1`] = `
<DocumentFragment>
  <div
    class="MuiCircularProgress-root pageProgress MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style="width: 40px; height: 40px;"
  >
    <svg
      class="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </div>
</DocumentFragment>
`;
